internal/osbuild2/xz_stage.go

```diff
@@ -0,0 +1,38 @@
+package osbuild2
+
+// Options for the org.osbuild.xz stage.
+type XZStageOptions struct {
+
+	Filename string `json:"filename"`
+}
+
+func (XZStageOptions) isStageOptions() {}
+
+type XZStageInput struct {
+	inputCommon
+	References XZStageReferences `json:"references"`
+}
+
+func (XZStageInput) isStageInput() {}
+
+type XZStageInputs struct {
+	File *XZStageInput `json:"file"`
+}
+
+func (XZStageInputs) isStageInputs() {}
+
+type XZStageReferences map[string]XZStageReference
+
+func (XZStageReferences) isReferences() {}
+
+type XZStageReference struct {
+	File string `json:"file"`
+}
```

Comment:
We're currently reworking the inputs because they shouldn't be stage-specific but have their own separate types.
Let's keep this for consistency with the existing inputs in composer.  Just making a note that this might change during the review of this PR.